Tests/AMPStorageTests.m

```diff
@@ -0,0 +1,135 @@
+    XCTAssertNotEqual([dir rangeOfString:expectedDir].location, NSNotFound);
+    
+    dir = [AMPStorage getDefaultIdentifyFile:NULL];
+    expectedDir = @"/Application Support/com.apple.dt.xctest.tool/$default_instance/amplitude_identify_storage.txt";
+    XCTAssertNotEqual([dir rangeOfString:expectedDir].location, NSNotFound);
+}
+
+- (void)testStoreEvent {
+    [AMPStorage storeEvent:exampleEvent instanceName:@"INSTANCE_NAME"];
+    NSString *eventsFilePath = [AMPStorage getDefaultEventsFile:@"INSTANCE_NAME"];
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    XCTAssertTrue([fileManager fileExistsAtPath:eventsFilePath]);
+    
+    NSString *content = [NSString stringWithContentsOfFile:eventsFilePath encoding:NSUTF8StringEncoding error:nil];
+    XCTAssertNotEqual([content rangeOfString:exampleEvent].location, NSNotFound);
+    XCTAssertNotEqual([content rangeOfString:@"{ \"batch\":"].location, NSNotFound);
+}
+
+- (void)testStoreIdentify {
+    [AMPStorage storeIdentify:exampleEvent instanceName:@"INSTANCE_NAME"];
+    NSString *identifyFilePath = [AMPStorage getDefaultIdentifyFile:@"INSTANCE_NAME"];
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    XCTAssertTrue([fileManager fileExistsAtPath:identifyFilePath]);
+    
+    NSString *content = [NSString stringWithContentsOfFile:identifyFilePath encoding:NSUTF8StringEncoding error:nil];
+    XCTAssertNotEqual([content rangeOfString:exampleEvent].location, NSNotFound);
+    XCTAssertNotEqual([content rangeOfString:@"{ \"batch\":"].location, NSNotFound);
+}
+
+/* This method also tests
+ + (void)start:(NSString *)path
+ + (void)finish:(NSString *)path
+ + (void)remove:(NSString *)path
+ */
+- (void)testStoreEventAndFinishAtUrl {
+    NSString *customPath = [AMPStorage getAppStorageAmpDir:@""];
+    customPath = [customPath stringByAppendingString:@"/custom_path/nested_path"];
+    NSURL *url = [NSURL fileURLWithPath:customPath];
+    [AMPStorage storeEventAtUrl:url event:exampleEvent];
+    [AMPStorage storeEventAtUrl:url event:exampleEvent];
+    
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    XCTAssertTrue([fileManager fileExistsAtPath:customPath]);
+    
+    [AMPStorage finish:customPath];
+    
+    NSString *content = [NSString stringWithContentsOfFile:customPath encoding:NSUTF8StringEncoding error:nil];
+    XCTAssertNotEqual([content rangeOfString:exampleEvent].location, NSNotFound);
+    XCTAssertNotEqual([content rangeOfString:@"{ \"batch\":"].location, NSNotFound);
+    
+    NSMutableArray *finishedFileDict = [AMPStorage getEventsFromDisk:customPath];
+    NSDictionary *secondEvent = finishedFileDict[1];
+    XCTAssertEqualObjects([secondEvent objectForKey:@"event_type"], @"testEvent");
+    
+    //Test a file that has not been finished
+    [AMPStorage remove:customPath];
+    XCTAssertFalse([fileManager fileExistsAtPath:customPath]);
+    
+    [AMPStorage storeEventAtUrl:url event:exampleEvent];
```

Comment:
Curious why we are test again here? Test the url can be reused again?

Reply:
It's testing the same filename rewritten again, but this time it does not have the method `finish` called on it.

Reply:
Gotcha, sounds fine for me